GameOfHive/HexagonGrid.swift

```diff
@@ -106,6 +115,76 @@ extension HexagonGrid: SequenceType {
     }
 }
 
+extension HexagonGrid: Encodable {
+    public func encode() -> JSON {
+        var data: [JSON] = []
+        for rowIndex in 0..<rows {
+            var rowString: String = ""
+            for columnIndex in 0..<columns {
+                if let hex = self.grid[rowIndex]?[columnIndex] {
+                    rowString += hex.active ? "1" : "0"
+                } else {
+                    rowString += "x"
+                }
+            }
+            data.append(.String(rowString))
+        }
+        let size: JSON = ["rows":rows,"columns":columns]
+        let grid: JSON = ["data":data,"size":size]
+        return ["grid":grid]
+    }
+}
+
+extension HexagonGrid: Decodable {
+    static var curriedInit: Int -> Int -> GridType -> HexagonGrid = { rows in
+        return { columns in
+            return { type in
+                return HexagonGrid(rows: rows, columns: columns, initialGridType: type)
+            }
+        }
+    }
+    
+    
+    
+    public static func decode(json: JSON) -> Decoded<HexagonGrid> {
+        let data: Decoded<[String]> = json <|| ["grid","data"]
+        let grid: Decoded<HexagonGrid> = curriedInit <^> json <| ["grid","size","rows"]
```

Comment:
kunnen we hier voor de leesbaarheid niet beter decodedRows en decodedGrid gebruiken?